packages/tables/frontend/sirius-components-tables/src/table/column/useTableColumnVisibility.ts

```diff
@@ -0,0 +1,54 @@
+/*******************************************************************************
+ * Copyright (c) 2024 CEA LIST.
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     Obeo - initial API and implementation
+ *******************************************************************************/
+import { useEffect, useState } from 'react';
+import { MRT_VisibilityState } from 'material-react-table';
+import { GQLTable } from '../TableContent.types';
+import { useTableMutations } from '../../graphql/mutation/useTableMutation';
+import { UseTableColumnVisibilityValue } from './useTableColumnVisibility.types';
+
+export const useTableColumnVisibility = (
+  editingContextId: string,
+  representationId: string,
+  table: GQLTable
+): UseTableColumnVisibilityValue => {
+  const [columnVisibility, setColumnVisibility] = useState<MRT_VisibilityState>(
+    table.columns.reduce((acc, obj) => {
+      acc[obj.id] = !obj.hidden;
+      return acc;
+    }, {})
+  );
```

Comment:
I don't understand why you need to have a state that needs to be updated and synchronized with useEffect if everything you do is to convert what the backend said. I wouldn't block the PR for that but `setColumnVisibility` could just perform the mutation and `columnVisibility` could simply be computed from table.

You would not have to try to reset the internal state after each refresh